### PR TITLE
feat: add transaction popup

### PR DESCRIPTION
**Pull request by im-adithya**
Uses [Credenza](https://github.com/redpangilinan/credenza)

## Desktop
Screenshot 2024-07-05 at 2 20 18 PM

## Mobile
Screenshot 2024-07-05 at 2 22 00 PM
Screenshot 2024-07-05 at 2 22 22 PM

### CODE REVIEW

File: frontend/src/components/TransactionsList.tsx [hunk 1]
```diff
@@ -28,79 +52,182 @@ function TransactionsList() {
...
+                    <div className="mt-8">
+                      <p className="dark:text-white">Date & Time</p>
+                      <p className="text-muted-foreground">
+                        {dayjs(tx.settled_at).toString()}
+                      </p>
+                    </div>
+                    <div className="mt-6">
+                      <p className="dark:text-white">Fee</p>
+                      <p className="text-muted-foreground">
+                        {tx.fees_paid} {tx.fees_paid == 1 ? "sat" : "sats"}
```
Comment (im-adithya) [inline frontend/src/components/TransactionsList.tsx:170]: In LND fees_paid is equivalent to tx.amount, have to check the backend before merging this.

File: frontend/package.json [hunk 2]
```diff
@@ -54,6 +54,7 @@
     "swr": "^2.2.4",
     "tailwind-merge": "^2.2.2",
     "tailwindcss-animate": "^1.0.7",
+    "vaul": "^0.9.1",
```
Comment (rolznz) [inline frontend/package.json:57]: @im-adithya how did you install it? did you do it through shadcn? there should be an update to the components json

File: frontend/src/components/TransactionsList.tsx (hunk 1 above)
Comment (rolznz) [reply frontend/src/components/TransactionsList.tsx:170]: we need a better way to do the sat/sats. Maybe we should start considering translations

File: frontend/src/components/TransactionsList.tsx [hunk 3]
```diff
@@ -28,79 +52,182 @@ function TransactionsList() {
...
+                      </div>
                     </div>
                   </div>
-                </div>
-              </div>
+                </CredenzaTrigger>
+                <CredenzaContent className="slashed-zero">
+                  <CredenzaHeader>
+                    <CredenzaTitle>
+                      {type == "outgoing" ? "Sent Bitcoin" : "Received Bitcoin"}
```
Comment (rolznz) [inline frontend/src/components/TransactionsList.tsx:127]: would it be possible to move this to a new component? this component is large already

File: frontend/package.json (hunk 2 above)
Comment (im-adithya) [reply frontend/package.json:57]: We use this in the [Drawer](https://ui.shadcn.com/docs/components/drawer) (See Manual installation option, I followed just that)

File: frontend/src/components/TransactionsList.tsx (hunk 1 above)
Comment (im-adithya) [reply frontend/src/components/TransactionsList.tsx:170]: I was thinking of that too, let's post on slack

File: frontend/package.json (hunk 2 above)
Comment (rolznz) [reply frontend/package.json:57]: hmm, I think the CLI makes some other changes, I think we should consistently use it? CC @reneaaron

Comment (im-adithya) [reply frontend/package.json:57]: Just tried using `npx shadcn-ui@latest add drawer`, it does nothing more but nicely does what's to be done manually

File: frontend/src/components/TransactionsList.tsx (hunk 3 above)
Comment (im-adithya) [reply frontend/src/components/TransactionsList.tsx:127]: Done!

File: frontend/src/components/TransactionItem.tsx [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+                  "w-6 h-6 md:w-8 md:h-8",
+                  type === "outgoing"
+                    ? "text-orange-400 dark:text-amber-600 stroke-orange-400 dark:stroke-amber-600"
+                    : "text-green-500 dark:text-emerald-500 stroke-green-400 dark:stroke-emerald-500"
+                )}
+              />
+            </div>
+            <div className="ml-4">
+              <p className="text-xl md:text-2xl font-semibold">
+                {Math.floor(tx.amount / 1000)}{" "}
```
Comment (rolznz) [inline frontend/src/components/TransactionItem.tsx:133]: can you format the number correctly? check the other places we show sat amounts

File: frontend/src/components/TransactionsList.tsx [hunk 5]
```diff
@@ -25,82 +24,8 @@ function TransactionsList() {
...
-                        <p className="text-xs text-gray-400 dark:text-neutral-600">
-                          ~{tx.totalAmountFiat}
-                        </p>
-                      )} */}
-                    </div>
-                  </div>
-                </div>
-              </div>
-            );
+            return <TransactionItem key={`tx-${i}`} tx={tx} />;
```
Comment (rolznz) [inline frontend/src/components/TransactionsList.tsx:27]: can you use the transaction payment hash as the key? using the index is kind of hacky and defeats the purpose

Comment (im-adithya) [reply frontend/src/components/TransactionsList.tsx:27]: Done!

**Review by rolznz — APPROVED** [review]
tACK

**Pull request merged**